google/auth/compute_engine/_metadata.py

```diff
@@ -202,22 +202,24 @@ def get(
 
     backoff = ExponentialBackoff(total_attempts=retry_count)
 
+    last_exception = None
     for attempt in backoff:
         try:
             response = request(url=url, method="GET", headers=headers_to_use)
             if response.status in transport.DEFAULT_RETRYABLE_STATUS_CODES:
-                _LOGGER.warning(
-                    "Compute Engine Metadata server unavailable on "
-                    "attempt %s of %s. Response status: %s",
-                    attempt,
-                    retry_count,
-                    response.status,
+                raise exceptions.TransportError(
```

Comment:
I think the retryable status code is expected in this case so I'm not sure a conversion to exception is the right thing and it makes the code a bit harder to understand.

(1) Can the customer set log level to an appropriate level to see these attempts?  This is the best solution because it preserves the full history of retry attempts.

(2) If the code gets here, `response` is set and you can just use that to raise the error below.  This would also catch the non-retryable error case.

Reply:
Please take another look at the code execution flow. It's likely that the code change is not changing the behavior the way you think it does. There are no new exceptions thrown to the user. And the logging still happens the same way as before. It's just that now in one of the cases instead of having duplicated logging code there the logging branch is triggered by an exception.

The initial solution was simple and did not change this area. However, @viacheslav-rostovtsev noticed an issue when the retriable error limit is exhausted. In that case the information about the service responses (retriable errors) is lost. A simplest way to capture that information in the final exception is to wrap it in TransportError and let it get assigned to `last_exception`.

Reply:
If we do need this change, then I agree with Wes's proposal in (2).

We could build the message `"Response status: {}; Response:\n{}".format(
                         response.status, response.data
                    ),` and store in a string, use it for the warning message and also use it in the final exception. The string will have the last response status. 

I do not understand why the execution flow needs to be changed to just log the last response.